modules/client/src/internal/client/methods.ts

```diff
@@ -1116,408 +1101,163 @@ export class ClientMethods extends ClientCore implements IClientMethods {
-      params: { where: { preparedSetupId } },
-      name: "pluginPreparation",
+  public async isPluginUpdate(
+    proposalId: string,
+  ): Promise<boolean> {
+    const name = "iproposal";
+    const query = QueryIProposal;
+    type T = { iproposal: SubgraphIProposal };
+    const { iproposal } = await this.graphql.request<T>({
+      query,
+      params: { id: proposalId.toLowerCase() },
+      name,
     });
-    if (pluginPreparation) {
-      // get the metadata of the plugin repo
-      // for the release and build specified
-      const release = pluginRepo.releases.find((
-        release: SubgraphPluginRepoRelease,
-      ) =>
-        release.release === decodedApplyUpdateActionParams.versionTag.release
-      );
-      const build = release?.builds.find((
-        build: { build: number; metadata: string },
-      ) => build.build === decodedApplyUpdateActionParams.versionTag.build);
-      const metadataCid = build?.metadata;
-
-      // fetch the metadata
-      const metadata = await this.ipfs.fetchString(metadataCid!);
-      const metadataJson = JSON.parse(metadata) as PluginRepoBuildMetadata;
-      // get the update abi for the specified build
-      const updateAbi = metadataJson.pluginSetup
-        .prepareUpdate[decodedApplyUpdateActionParams.versionTag.build]?.inputs;
-      if (updateAbi) {
-        // if the abi exists try to decode the data
-        try {
-          if (
-            decodedApplyUpdateActionParams.initData.length > 0 &&
-            updateAbi.length === 0
-          ) {
-            throw new Error();
-          }
-          // if the decode does not throw an error the data is valid
-          defaultAbiCoder.decode(
-            getNamedTypesFromMetadata(updateAbi),
-            decodedApplyUpdateActionParams.initData,
-          );
-        } catch {
-          // if the decode throws an error the data is invalid
-          causes.push(
-            PluginUpdateProposalInValidityCause.INVALID_DATA,
-          );
-        }
-      } else {
-        causes.push(
-          PluginUpdateProposalInValidityCause.INVALID_PLUGIN_REPO_METADATA,
-        );
-      }
-    } else {
-      causes.push(
-        PluginUpdateProposalInValidityCause.MISSING_PLUGIN_PREPARATION,
-      );
+    if (!iproposal) {
+      return false;
     }
-    return causes;
+    const subgraphActions = iproposal.actions;
+    let actions = toDaoActions(subgraphActions);
+    const classifiedActions = classifyProposalActions(actions);
+    return isPluginUpdateAction(classifiedActions) ||
+      isPluginUpdateActionWithRootPermission(classifiedActions);
   }
-
   /**
    * Check if the specified proposal id is valid for updating a plugin
-   * The failure map should be checked before calling this method
    *
-   * @param {DaoAction[]} actions
+   * @param {string} proposalId
    * @return {*}  {Promise<PluginUpdateProposalValidity>}
    * @memberof ClientMethods
    */
   public async isPluginUpdateValid(
```

Comment:
```suggestion
  public async isPluginUpdateProposalValid(
```
Reading the name/jsdoc alone, I ask myself if it validates a single action, a block, or the entire proposal.

Reply:
I left as 

`isPluginUpdateProposal` for the boolean
`isPluginUpdateProposalValid` foir the full check

for me is clear enough, we can give a second thought if you feel is necessary